langchain/utilities/bash.py

```diff
@@ -0,0 +1,24 @@
+import subprocess
+from typing import List
+
+class BashProcess:
+    """Executes bash commands and returns the output."""
+
+    def __init__(self, strip_newlines: bool = False):
+        self.strip_newlines = strip_newlines
+
+
+    def run(self, commands: List[str]) -> List[str]:
+        outputs = []
+        for command in commands:
+            try:
+                output = subprocess.check_output(command, shell=True).decode()
+                if self.strip_newlines:
+                    output = output.strip()
+                outputs.append(output)
+            except subprocess.CalledProcessError as error:
+                outputs.append(str(error))
+                return {"success": False, "outputs": outputs}
```

Comment:
do we use the success anywhere? slash should we

Reply:
If we have multiple bash commands, being run one after another, there's a possibility it could fail partway. So we don't need to have success here but then the LLM chain user needs to parse the output to look for an error. That sounded trickier to me than checking for success and then grabbing the outputs